### ISSUE TITLE
Incorrectly Identifying Binary Variable as Continuous (unhelpful error)

### ISSUE THREAD

**Issue by akgold** [anniejw6/modmarg#72]
I thought we had changed so everything didn't have to be cast to factor. If we didn't need more helpful error.

Error: `Error in pred_se(df_trans = x, var_interest = var_interest, model = mod, : We do not support effects for continuous variables at this time.`

```{r}
data(margex)

g <- glm(y ~ poly(age, 2)*treatment , data = margex)
summary(g)
modmarg::mod_marg2(g, var_interest = "treatment", type = "effects")

```

**Comment by anniejw6** [comment]
`margex$treatment` is an integer here, which is why it thinks it's continuous. 

This works fine:

```
> data(margex)
> margex$treatment <- as.character(margex$treatment)
> g <- glm(y ~ poly(age, 2)*treatment , data = margex)
> modmarg::mod_marg2(g, var_interest = "treatment", type = "effects")
[[1]]
          Label   Margin Standard.Error Test.Stat      P.Value Lower CI (95%)
1 treatment = 0  0.00000      0.0000000       NaN          NaN        0.00000
2 treatment = 1 14.17061      0.7802453  18.16174 5.201016e-70       12.64136
  Upper CI (95%)
1        0.00000
2       15.69987
```

**Comment by akgold** [comment]
As does the factor version, which is what I ran after error -- in either case, think we need a more helpful error. I can already envision someone smashing their head into the screen "BUT IT'S AN INTEGER!!!"

**Comment by nathanielolin** [comment]
Addressed in PR #78
